### PR TITLE
imported wallet password form and split it

### PR DESCRIPTION
**Pull request by Maxstgt**

### CODE REVIEW

File: libs/ethers/src/lib/wallet/forms/create-key-password-form.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {
+  EntityControl,
+  PasswordControl,
+  FormEntity
+} from '@blockframes/utils';
+
+interface CreateKeyPassword {
+  password: string
+}
+function createKeyPassword(params?: Partial<CreateKeyPassword>): CreateKeyPassword {
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/forms/create-key-password-form.ts]: This is a one key object. Why do you need this function ?

File: libs/ethers/src/lib/wallet/forms/create-key-password-form.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import {
+  EntityControl,
+  PasswordControl,
+  FormEntity
+} from '@blockframes/utils';
+
+interface CreateKeyPassword {
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/forms/create-key-password-form.ts]: What the point of this interface ?

File: libs/ethers/src/lib/wallet/forms/create-key-password-form.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+function createKeyPasswordControls(entity: Partial<CreateKeyPassword>): EntityControl<CreateKeyPassword> {
+  const createPasswordKey = createKeyPassword(entity);
+  return {
+    password: new PasswordControl(createPasswordKey.password),
+  }
+}
+
+type CreateKeyPasswordControl = ReturnType<typeof createKeyPasswordControls>;
+
+export class CreateKeyPasswordForm extends FormEntity<CreateKeyPassword, CreateKeyPasswordControl> {
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/forms/create-key-password-form.ts]: This is a one key form. You don't need to create a whole `FormEntity` for that. Just use a `ControlForm('')` when you need it.

File: libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.html [hunk 4]
```diff
@@ -1,34 +1,37 @@
...
   <section *ngSwitchCase="steps.password">
-    <ng-container *ngIf="!(loading$ | async) as loading; else isLoading">
-      <!-- TODO issue 557 integrate Max's components -->
-      <h2>Create a password</h2>
-      <p>This is a mock component, it will be replaced by "wallet-password-form"</p>
-      <button mat-raised-button color="accent" (click)="setPassword('azerty')">Create</button>
-      <!-- <wallet-password-form [passwordView]="false" (password)="setPassword($event)"></wallet-password-form> -->
-      <!-- TODO -->
+    <ng-container *ngIf="!(loading$ | async) as loading; else: isLoading">
+      <wallet-create-password-form
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.html]: Format on one line.

File: libs/ethers/src/lib/wallet/forms/create-key-password-form.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  constructor(data?: CreateKeyPassword) {
+    super(
+      createKeyPasswordControls(data)
+    )
+  }
+}
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/forms/create-key-password-form.ts]: I think you don't need this file.

File: libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.html [hunk 6]
```diff
@@ -1,34 +1,37 @@
...
     </p>
     <a mat-button [href]="jsonKeystore(true)" [download]="keyName">Download</a>
     <button mat-raised-button color="accent" (click)="step = steps.import">Next</button>
   </section>
   <section *ngSwitchCase="steps.import">
-    <wallet-import-key-form (importKey)="setKey($event)" [lockMnemonic]="true"></wallet-import-key-form>
-    <button mat-button (click)="step = steps.EXPORT">Previous</button>
+    <wallet-import-key-form
+      (importKey)="setKey($event)"
+      [lockMnemonic]="true"
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.html]: Put input `[lockMnemonic]` before output `(importKey)`.

Comment (GrandSchtroumpf) [reply libs/ethers/src/lib/wallet/wallet-add-key/wallet-add-key.component.html]: Also here, `lockMnemonic` is a boolean value (always true) so you should do `<wallet-import-key-form lockMnemonic>`.
Here is the rule : 
- Attribute is a string : `<div attr="mystring">`
- Attribute is a boolean : `<div attr>`
- Attribute is a javascript value : `<div [attr]="myValue">`

File: libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<section fxLayout="row" fxLayoutAlign="center" fxLayoutGap="80px">
+  <img
+    src="/assets/images/key_lock_laptop.png"
+    alt="Laptop which is protected by a lock"
+  />
+  <article fxLayout="column" fxLayoutAlign="center center">
+    <h1>Unlock your key</h1>
+    <p>
+      You know, it really doesn’t matter what you write as long as you’ve got a young, and
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.html]: What is that text ? This is the best way to never update it until is too last. We are used to Lorem Ipsum. If you don't see a text starting with Lorem Ipsum, you think this is the final one.
So either you have the final content, or you put Lorem Ipsum.

File: libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.html [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      pounds. Ok?
+    </p>
+    <mat-form-field appearance="outline">
+      <input matInput type="password" placeholder="Password" [formControl]="password" required />
+      <mat-hint>Password is required</mat-hint>
+      <mat-error *ngIf="password.hasError('required')">You must type in your password</mat-error>
+      <mat-error *ngIf="password.hasError('maxlength')"
+        >Password with more than <strong>24</strong> characters.</mat-error
+      >
+      <mat-error *ngIf="password.hasError('minlength')"
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.html]: Same here

File: libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.scss [hunk 9]
```diff
@@ -0,0 +1,5 @@
+img {
+  width: 410px;
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.scss]: Have you tried in different size of screen ?

File: libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.scss [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+  height: 435px;
+  user-select: none;
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.scss]: Why ?

File: libs/ethers/src/lib/wallet/wallet-create-password-form/wallet-create-password-form.component.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+export class WalletCreatePasswordFormComponent {
+  public confirmPasswordForm = new CreateKeyPasswordForm();
+
+  /** An event that will send the setted password by the user */
+  @Output() userPassword: EventEmitter<string> = new EventEmitter();
+
+  constructor() {}
+
+  public submitPassword(){
+    this.userPassword.emit(this.confirmPasswordForm.get('password').value);
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-create-password-form/wallet-create-password-form.component.ts]: You'll need to update that after changing to a simple PasswordControl.

File: libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html [hunk 12]
```diff
@@ -1,29 +1,36 @@
...
     </ng-template>
   </section>
-  <img src="/assets/images/import_key_pic.png">
-  <div>
-    <button mat-raised-button color="primary" (click)="import()">Import your Key</button>
-  </div>
-</main>
\ No newline at end of file
+  <img src="/assets/images/import_key_pic.png" />
+  <div><button mat-raised-button color="primary" (click)="import()">Import your Key</button></div>
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html]: This div is useless. Remove it.

File: libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html [hunk 13]
```diff
@@ -1,29 +1,36 @@
...
+      ></file-upload>
     </ng-container>
     <ng-template #mnemonicImport>
-
       <mat-form-field>
-        <textarea matInput placeholder="Enter or paste your mnemonic" [formControl]="mnemonicField"></textarea>
+        <textarea
+          matInput
+          placeholder="Enter or paste your mnemonic"
+          [formControl]="mnemonicField"
```
Comment (GrandSchtroumpf) [inline libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html]: I think the name of the control should be "mnemonic". The "field" doesn't provide much more information.

File: libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html (hunk 12 above)
Comment (pldespaigne) [reply libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html]: this is done in #567

File: libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html (hunk 13 above)
Comment (pldespaigne) [reply libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html]: this is done in #567

Comment (Maxstgt) [reply libs/ethers/src/lib/wallet/wallet-import-key-form/wallet-import-key-form.component.html]: it has to be 'field' cause inside of the TS file there is already mnemonic variable

File: libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.html (hunk 7 above)
Comment (GrandSchtroumpf) [reply libs/ethers/src/lib/wallet/wallet-ask-password-form/wallet-ask-password-form.component.html]: Lorem ipsum is as long as you need. You can generate one [here](https://www.lipsum.com/) if you need a whole page of text.

**Pull request merged**